OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -48,10 +48,16 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised)
+					return self.Owner;
 
-				return self.Owner;
+				Player p = null;
+				if (self.World.LocalPlayer != null && self.World.LocalPlayer.WinState == WinState.Undefined)
```

Comment:
Yep, you're right. Argh, so many special cases!

It would have to be
```csharp
var p = self.World.RenderPlayer ?? self.World.LocalPlayer;
if (p == null || p.Spectating || self.World.RenderPlayer == null)
        // return spectator stuff
else
     return p.IsAlliedWith(self.Owner) ? // <stuff for allies> : <stuff for enemies>
```
then if you wanted to do that. That looks kinda ugly now, though. Surely there's a way to express those conditions better, but I'm not seeing it atm.